packages/block-editor/src/components/block-toolbar/style.scss

```diff
@@ -84,6 +88,12 @@
 				display: none;
 			}
 		}
+
+		// These rules ensure that icons are always positioned in a way that lines up with the rest of the icons in the toolbar.
+		.selected-block-tools-wrapper & {
+			height: $header-height;
+			padding-top: 7px; // It should be 6px (60px header height - 48px toolbar height = 12 / 2), but there is a -1px top-margin down the stack that affects this.
+		}
```

Comment:
Any rules related to the block toolbar within the header should be moved into the `header.scss` files for that package (edit-post, edit-site, etc), as the `block-toolbar/style.scss` shouldn't "know about" how it's used outside of the block editor.